programs/bubblegum/src/lib.rs

```diff
@@ -332,10 +344,14 @@ pub mod bubblegum {
     }
 
     pub fn mint(ctx: Context<Mint>, version: Version, message: MetadataArgs) -> Result<()> {
+        // TODO -> Pass collection in check collection authority or collection delegate authority signer
+        // TODO -> Separate V1 / V1 into seperate instructions
+        //
         let owner = ctx.accounts.owner.key();
         let delegate = ctx.accounts.delegate.key();
         let merkle_slab = ctx.accounts.merkle_slab.to_account_info();
         let nonce = &mut ctx.accounts.nonce;
+        let asset_id = ctx.accounts.asset_id.key();
```

Comment:
Why does the asset_id need to be passed in the instruction at all? I think it should be `FPA(tree_id, authority, token_program)`. We can log the hash and store it in the indexer